### PR TITLE
kubelet: remove `--container-runtime` command line argument

### PR DESCRIPTION
**Pull request by aimuz**
Signed-off-by: aimuz <mr.imuz@gmail.com>

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

xref #106893

#### Does this PR introduce a user-facing change?

```release-note
kubelet: remove already-deprecated `--container-runtime` command line argument
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sftim** [comment]
Changelog text suggestion (adding backticks, and avoiding Go-specific jargon for command line argument)
```markdown
kubelet: remove `--container-runtime` command line argument
```
or
```markdown
kubelet: remove already-deprecated `--container-runtime` command line argument
```

**Comment by endocrimes** [comment]
+1 on updating the changelog to either of tims suggestions

**Comment by aimuz** [comment]
Thanks, I made the adjustment

**Comment by aimuz** [comment]
rebase

File: cmd/kubelet/app/options/options.go [hunk 1]
```diff
@@ -185,10 +184,6 @@ func ValidateKubeletFlags(f *KubeletFlags) error {
 		return fmt.Errorf("the SeccompDefault feature gate must be enabled in order to use the --seccomp-default flag")
 	}
 
-	if f.ContainerRuntime != kubetypes.RemoteContainerRuntime {
-		return fmt.Errorf("unsupported CRI runtime: %q, only %q is currently supported", f.ContainerRuntime, kubetypes.RemoteContainerRuntime)
-	}
-
```
Comment (SataQiu) [inline cmd/kubelet/app/options/options.go:191]: Hi @aimuz 
Line 486 ~ 487 also need to be updated.
>
	fs.StringVar(&c.ContainerLogMaxSize, "container-log-max-size", c.ContainerLogMaxSize, "<Warning: Beta feature> Set the maximum size (e.g. 10Mi) of container log file before it is rotated. This flag can only be used with --container-runtime=remote.")
	fs.Int32Var(&c.ContainerLogMaxFiles, "container-log-max-files", c.ContainerLogMaxFiles, "<Warning: Beta feature> Set the maximum number of container log files that can be present for a container. The number must be >= 2. This flag can only be used with --container-runtime=remote.")

Comment (aimuz) [reply cmd/kubelet/app/options/options.go:185]: Fixed

**Comment by aimuz** [comment]
rebase

**Comment by aimuz** [comment]
repeat 114017

**Pull request closed without merging**